File: backend/app/schemas/candidate.py
"""Candidate schemas."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class CandidateCreate(BaseModel):
    name: str
    headline: str | None = None
    location: str | None = None
    phone: str | None = None
    bio: str | None = None
    preferred_roles: list[str] | None = None
    preferred_locations: list[str] | None = None
    open_to_remote: bool = True
    salary_expectation_min: int | None = None
    salary_expectation_max: int | None = None
    salary_currency: str = "USD"
    availability: str | None = None


class CandidateUpdate(BaseModel):
    name: str | None = None
    headline: str | None = None
    location: str | None = None
    phone: str | None = None
    bio: str | None = None
    current_role: str | None = None
    current_company: str | None = None
    preferred_roles: list[str] | None = None
    preferred_locations: list[str] | None = None
    open_to_remote: bool | None = None
    salary_expectation_min: int | None = None
    salary_expectation_max: int | None = None
    salary_currency: str | None = None
    availability: str | None = None


class CandidateResponse(BaseModel):
    id: str
    user_id: str
    name: str
    headline: str | None = None
    location: str | None = None
    phone: str | None = None
    bio: str | None = None
    years_of_experience: Decimal | None = None
    current_role: str | None = None
    current_company: str | None = None
    preferred_roles: list[str] | None = None
    preferred_locations: list[str] | None = None
    open_to_remote: bool = True
    salary_expectation_min: int | None = None
    salary_expectation_max: int | None = None
    salary_currency: str = "USD"
    availability: str | None = None
    profile_completeness: int = 0
    evidence_confidence: str = "none"
    profile_summary: dict | None = None
    overall_scores: dict | None = None
    last_analyzed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CandidateListResponse(BaseModel):
    id: str
    name: str
    headline: str | None = None
    location: str | None = None
    years_of_experience: Decimal | None = None
    current_role: str | None = None
    evidence_confidence: str = "none"
    profile_completeness: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/candidate.ts
export interface Candidate {
  id: string;
  user_id: string;
  name: string;
  headline?: string;
  location?: string;
  phone?: string;
  bio?: string;
  years_of_experience?: number;
  current_role?: string;
  current_company?: string;
  preferred_roles?: string[];
  preferred_locations?: string[];
  open_to_remote: boolean;
  salary_expectation_min?: number;
  salary_expectation_max?: number;
  salary_currency: string;
  availability?: string;
  profile_completeness: number;
  evidence_confidence: string;
  profile_summary?: Record<string, unknown>;
  overall_scores?: Record<string, number>;
  last_analyzed_at?: string;
  created_at: string;
  updated_at: string;
}

export interface Skill {
  id: string;
  candidate_id: string;
  name: string;
  category?: string;
  confidence: string;
  evidence_sources?: Record<string, unknown>;
  proficiency_level?: string;
  verified: boolean;
}

export interface Resume {
  id: string;
  candidate_id: string;
  file_url: string;
  file_name?: string;
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  processing_status: string;
  is_primary: boolean;
  analyzed_at?: string;
  created_at: string;
}

export interface GitHubProfile {
  id: string;
  candidate_id: string;
  username: string;
  profile_url?: string;
  public_repos?: number;
  followers?: number;
  total_stars?: number;
  primary_languages?: Record<string, number>;
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  processing_status: string;
  analyzed_at?: string;
}

export interface Portfolio {
  id: string;
  candidate_id: string;
  url: string;
  title?: string;
  portfolio_type?: string;
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  processing_status: string;
}

export interface Project {
  id: string;
  candidate_id: string;
  title: string;
  description?: string;
  role?: string;
  technologies?: string[];
  url?: string;
  github_url?: string;
  live_url?: string;
  start_date?: string;
  end_date?: string;
  is_ongoing: boolean;
  scope?: string;
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  processing_status: string;
  created_at: string;
}

export interface Certificate {
  id: string;
  candidate_id: string;
  name: string;
  issuer?: string;
  issue_date?: string;
  credential_url?: string;
  difficulty_level?: string;
  skills_covered?: string[];
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  is_verified: boolean;
  processing_status: string;
}

export interface Video {
  id: string;
  candidate_id: string;
  title?: string;
  file_url: string;
  duration_seconds?: number;
  analysis?: Record<string, unknown>;
  scores?: Record<string, number>;
  processing_status: string;
}

export interface WorkHistory {
  id: string;
  candidate_id: string;
  company: string;
  title: string;
  location?: string;
  start_date?: string;
  end_date?: string;
  is_current: boolean;
  employment_type?: string;
  description?: string;
  responsibilities?: string[];
  achievements?: string[];
  technologies?: string[];
  analysis?: Record<string, unknown>;
}
